### PR TITLE
Update Doc string for SnowflakeOperatorAsync

### PR DESCRIPTION
**Pull request by bharanidharan14**
Added Doc string for the SnowflakeOperatorAsync to get to know the use of this operator compared with snowflake sql API operator, best practices to be followed while using this operator and where can this `SnowflakeOperatorAsync` can be used

closes: #557
Screenshot 2022-07-28 at 2 56 48 PM
Screenshot 2022-07-28 at 5 56 44 PM

### CODE REVIEW

**Comment by bharanidharan14** [comment]
@phanikumv  @rajaths010494 @pankajastro @pankajkoti, can you review this PR

**Comment by pankajastro** [comment]
> @phanikumv @rajaths010494 @pankajastro @pankajkoti, can you review this PR

Can you please post the screenshot like how it is looking you can run `make docs`?

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajastro) [inline astronomer/providers/snowflake/operators/snowflake.py:33]: see also directive in between docs string creates a weird big block.

**Comment by rajaths010494** [comment]
LGTM , once Screenshot is added we can close.

**Comment by bharanidharan14** [comment]
@rajaths010494 @pankajastro Added screenshot

**Comment by pankajastro** [comment]
> @rajaths010494 @pankajastro Added screenshot

I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me 
> for the query status based on the query id from different connection.
    .. see also::
        https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.

**Comment by bharanidharan14** [comment]
> > @rajaths010494 @pankajastro Added screenshot
> 
> I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me
> 
> > for the query status based on the query id from different connection.
> > .. see also::
> > https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.

See also link is not visible in Docs, its not showing up in any of the operators/sensors

**Comment by bharanidharan14** [comment]
> > > > @rajaths010494 @pankajastro Added screenshot
> > > 
> > > 
> > > I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me
> > > > for the query status based on the query id from different connection.
> > > > .. see also::
> > > > https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.
> > 
> > 
> > See also link is not visible in Docs, its not showing up in any of the operators/sensors
> 
> then this is a bug in our documentation please create a ticket to track it. But I was able to do it in #464

Created Ticket

**Review by pankajastro — APPROVED** [review]

**Review by rajaths010494 — APPROVED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:32]:
```suggestion
    Where can this operator fit in?
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:33]:
```suggestion
```

Comment (phanikumv) [reply astronomer/providers/snowflake/operators/snowflake.py:33]: Remove above as it is already covered in previous point

**Review by phanikumv — CHANGES_REQUESTED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:33]: Removed this line
`Whether the single query or multiple query to execute the query in parallel and free up the worker resource.`

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:32]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:24]: Changed the first letter to caps

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Addressed as per suggested comment

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

**Comment by bharanidharan14** [comment]
@phanikumv Addressed your PR review changes

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:162]:
```suggestion
    Where can this operator fit in?
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:163]:
```suggestion
         - To execute multiple SQL statements in a single request
```

**Review by phanikumv — CHANGES_REQUESTED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:163]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:162]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:22]:
```suggestion
    - SnowflakeOperatorAsync uses the snowflake python connector ``execute_async`` method to submit a database command
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:25]:
```suggestion
    - Accepts list of queries or multiple queries with ‘;’ semicolon separated string and params. It loops through the
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:26]:
```suggestion
      queries and executes them in sequence. Uses execute_async method to run the query
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:29]:
```suggestion
    - The trigger gets the list of query IDs as input and polls every few seconds to snowflake and checks
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:30]:
```suggestion
      for the query status based on the query ID from different connection.
```

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:31]: Do we need `-` for the above statements? I guess we can remove those.

**Review by pankajkoti — APPROVED** [review]
LGTM, minor suggestions.

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:31]: Document doesn't look good Without `-`, Initially I added as paragraph it looks congested

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:30]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:29]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:27]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:26]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

**Comment by bharanidharan14** [comment]
@phanikumv Addressed your review comments, can you please review it again

**Comment by bharanidharan14** [comment]
@phanikumv Need your approval

**Review by phanikumv — APPROVED** [review]

**Pull request merged**
